src/rest.rs

```diff
@@ -424,11 +438,11 @@ impl<'a> Channel<'a> {
     ///
     /// Returns a history::RequestBuilder which is used to set parameters
     /// before sending the history request.
-    pub fn history(&self) -> history::PaginatedRequestBuilder<Message> {
-        self.rest.paginated_request(
+    pub fn history(&self) -> PaginatedRequestBuilder<Message> {
+        self.rest.paginated_request_with_handler(
```

Comment:
It seems like every call to `self.rest.paginated_request` has become `self.rest.paginated_request_with_handler` in this PR, so perhaps just change the signature of `self.rest.paginated_request`?

Reply:
paginated_request_with_handler is the special function that lets you do pass in some extra data, while paginated_request is more basic but probably want most of the time.

Internally paginated_request_with_handler is used more as messages require the decode step. But these functions meant to satisfy the request() spec point. So even though the other function isn't used by use, it's meant to be used by the user,

Also one is implemented in terms of the other, there's no code duplication.